crates/sui-core/src/consensus_handler.rs

```diff
@@ -155,11 +155,18 @@ impl<T: ObjectStore + Send + Sync, C: CheckpointServiceNotify + Send + Sync> Exe
             .index
             .last_committed_round;
 
+        let round = consensus_output.sub_dag.leader_round();
+
+        assert!(round >= last_committed_round);
+        if last_committed_round == round {
+            // we can receive the same commit twice under certain circumstances.
```

Comment:
I am not allowed to comment on line [419 directly](https://github.com/MystenLabs/sui/pull/13819/files#diff-8ccde7467e1b7899685cdb5469dc530bc63b1945339660881ad1267287d4e2e7R419) so writing the comment here...it would it be possible to have a crash before line 419 and after processing the transactions (having updated the highest commit index) effectively leading to missing creating a checkpoint after recovery since we won't re-sequence now again the last sub dag? I am definitely missing info around checkpoints (maybe next commit would backfill that info - not aware), but thinking what the repercussions would be here.

Reply:
will add the log - I think the rest of this is taken care of now that the handler is atomic